### PR TITLE
added the OpenShift Template for the Argo Workflow

### PR DESCRIPTION
**Pull request by goern**
Signed-off-by: Christoph Görn <goern@redhat.com>

### CODE REVIEW

File: openshift/solve_and_sync-template.yaml [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+          value: "${THOTH_SOLVER_WORKFLOW_ID}"
+        - name: "THOTH_LOG_SOLVER"
+          value: "${THOTH_LOG_SOLVER}"
+        - name: "THOTH_SOLVER_NO_TRANSITIVE"
+          value: "${THOTH_SOLVER_NO_TRANSITIVE}"
+        - name: "THOTH_SOLVER_PACKAGES"
+          value: "${THOTH_SOLVER_PACKAGES}"
+        - name: "THOTH_SOLVER_INDEXES"
+          value: "${THOTH_SOLVER_INDEXES}"
+        - name: "THOTH_FORCE_SYNC"
```
Comment (CermakM) [inline openshift/solve_and_sync-template.yaml]: Missing from the template parameters

**Comment by goern** [comment]
@CermakM PTAL

File: openshift/solve_and_sync-template.yaml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+                  name: "graph-sync"
+                  template: "solver"
+                arguments:
+                  artifacts:
+                    - name: solver-document
+                      from: "{{tasks.solverrhel80py36.outputs.artifacts.solver-document}}"
+                  parameters:
+                    - name: "THOTH_SOLVER_DOCUMENT_ID"
+                      value: >
+                        "solver-{{workflow.parameters.THOTH_SOLVER_NAME}}-{{workflow.parameters.THOTH_SOLVER_WORKFLOW_ID}}"
```
Comment (CermakM) [inline openshift/solve_and_sync-template.yaml]: Use `>-`, no quotes

File: thoth/solver/python/instrument.py [hunk 3]
```diff
@@ -98,7 +98,7 @@ def _get_importlib_metadata_metadata(package_name):  # type: (str) -> None
 
 
 def _get_importlib_metadata_version(package_name):  # type: (str) -> None
-    """"Retrieve version based for the given package."""
+    """Retrieve version based for the given package."""
```
Comment (CermakM) [inline thoth/solver/python/instrument.py]: "based *on*"

File: thoth/solver/python/instrument.py [hunk 4]
```diff
@@ -110,7 +110,7 @@ def _get_importlib_metadata_version(package_name):  # type: (str) -> None
 
 
 def _get_importlib_metadata_requires(package_name):  # type: (str) -> None
-    """"Retrieve requires based for the given package."""
+    """Retrieve requires based for the given package."""
```
Comment (CermakM) [inline thoth/solver/python/instrument.py]: "based *on*"

File: openshift/solve_and_sync-template.yaml [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+            tasks:
+              - name: "solverrhel80py36"
+                templateRef:
+                  name: "solver-rhel-8.0-py36"
+                  template: "solve"
+                arguments:
+                  parameters:
+                    - name: "THOTH_SOLVER_DOCUMENT_ID"
+                      value: >
+                        "solver-{{workflow.parameters.THOTH_SOLVER_NAME}}-{{workflow.parameters.THOTH_SOLVER_WORKFLOW_ID}}"
```
Comment (CermakM) [inline openshift/solve_and_sync-template.yaml]: Use `>-`, no quotes

File: openshift/solve_and_sync-template.yaml [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+    description: "Log solver actions"
+    displayName: "Log solver actions"
+    required: true
+    value: INFO
+
+objects:
+  - apiVersion: argoproj.io/v1alpha1
+    kind: Workflow
+    metadata:
+      generateName: solver-
```
Comment (CermakM) [inline openshift/solve_and_sync-template.yaml]: We might want to differentiate the name of the workflow with the `THOTH_SOLVER_WORKFLOW_ID` as well?

**Pull request merged**